sdk/appconfiguration/app-configuration/test/public/secretReference.spec.ts

```diff
@@ -163,4 +163,25 @@ describe("AppConfigurationClient - SecretReference", () => {
       await client.deleteConfigurationSetting({ key: secondSetting.key });
     });
   });
+
+  describe("serializeAsConfigurationSettingParam", () => {
+    [`[]`, "Hello World"].forEach((value) => {
```

Comment:
Question: is `[]` here to check something specific (like are you just ensuring that just because something is kind of 'JSON-ish' that it's not used that way)?

Reply:
These tests are straight from https://github.com/Azure/azure-sdk/issues/2801.

The motivation behind these is to make sure the updates that we have done are not breaking changes.
Since these examples though special settings as per the content type, they can't be encoded.
Since the previous versions support such cases, we should not break those users.